### PR TITLE
General bug fixing

### PR DESCRIPTION
**Pull request by PBWiseman**
## Describe your changes

Making general bug fixes

Completed:
- Full Breakfast fixed to be 30% HP instead of 3% HP
- Lucky Dive Max HP now scales with health items
- Sword of the Master beams now damage Peacekeeper's shield
- Sword swings now damage Peacekeeper's shield
- Reflections work on boss bullets
- Spine Plate can reflect Peacekeeper bullets
- Peacekeeper will now always pick his gun back up after his shield is broken

## Issue number and link

#146 

## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: Assets/Scripts/Player/Sword.cs [hunk 1]
```diff
@@ -38,7 +42,14 @@ void OnTriggerEnter2D(Collider2D other)
...
             bulletInstance.GetComponent<Rigidbody2D>().velocity = -other.gameObject.GetComponent<Rigidbody2D>().velocity;
-            bulletInstance.GetComponent<ReflectedBullet>().Damage = other.gameObject.GetComponent<EnemyBullet>().Damage;
+            if (other.gameObject.GetComponent<EnemyBullet>() != null)
+            {
+                bulletInstance.GetComponent<ReflectedBullet>().Damage = other.gameObject.GetComponent<EnemyBullet>().Damage;
+            }
+            else if (other.gameObject.GetComponent<BossBullet>() != null)
+            {
+                bulletInstance.GetComponent<ReflectedBullet>().Damage = other.gameObject.GetComponent<BossBullet>().BulletDamage;
+            }
```
Comment (AlexReidNZ) [inline Assets/Scripts/Player/Sword.cs:52]: maybe add a final else here, setting the damage to a base number if an attack is reflected that doesn't have a set damage value

**Review by AlexReidNZ — CHANGES_REQUESTED** [review]
Tested all the bugs listed on the PR, they are all fixed correctly. There is another issue with the reflector now, you can reflect the peacekeepers napalm bombs, but the reflected attack does no damage. Maybe give the deflected bullet a small amount of damage that it uses if it's damage is set to 0.

File: Assets/Scripts/Player/Sword.cs (hunk 1 above)
Comment (PBWiseman) [reply Assets/Scripts/Player/Sword.cs:53]: Made napalm bombs (and regular bombs) no longer get reflected

**Review by AlexReidNZ — APPROVED** [review]
Requested fixes have been implemented, looks good for merging now.

**Review by kblack89 — APPROVED** [review]
Played through and tested this, no apparent bugs and code is clean and makes sense.

**Pull request merged**